lib/charms/opensearch/v0/models.py

```diff
@@ -16,13 +20,56 @@
 LIBPATCH = 1
 
 
-class Node:
+class Model(ABC, BaseModel):
+    """Base model class."""
+
+    def to_str(self) -> str:
+        """Deserialize object into a string."""
+        return self.json()
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Deserialize object into a dict."""
+        return self.dict()
```

Comment:
I did it temporarily to hide the pydantic methods, as the APIs seem to be different in Pydantic `1.x` vs `2.x`.  I could not use the v2 because of some rust dependency issues - but the plan would be to move away from 1.x and then get rid of this

Reply:
Oh, I see. In this case it is fine I suppose!